### PR TITLE
s/float/double/ for GPUColor to support i32/u32.

### PR DESCRIPTION
**Pull request by jdashg**
***
<a href="https://pr-preview.s3.amazonaws.com/jdashg/gpuweb/pull/355.html" title="Last updated on Jul 12, 2019, 9:35 PM UTC (5a22e7f)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/gpuweb/gpuweb/355/c6db522...jdashg:5a22e7f.html" title="Last updated on Jul 12, 2019, 9:35 PM UTC (5a22e7f)">Diff</a>

### CODE REVIEW

File: spec/index.bs [hunk 1]
```diff
@@ -24,12 +24,12 @@ typedef unsigned long u32;
...
-    required float g;
-    required float b;
-    required float a;
+    required double r; // `double` is large enough for precise i32/u32.
+    required double g;
+    required double b;
+    required double a;
 };
-typedef (sequence<float> or GPUColorDict) GPUColor;
+typedef (sequence<double> or GPUColorDict) GPUColor;
```
Comment (kvark) [inline spec/index.bs]: did you consider the alternative of explicitly supporting integers, i.e
```
dictionary GPUColorFloatDict {
    required float r;
    required float g;
    required float b;
    required float a;
};
dictionary GPUColorIntDict {
    required i32 r;
    required i32 g;
    required i32 b;
    required i32 a;
};
typedef (sequence<float> or GPUColorFloatDict or sequence<i32> or GPUColorIntDict) GPUColor;
```

It seems a bit strange to try to fit integers into doubles if we already have `GPUColor` defined as an enum (of sequence or a dict).

**Comment by jdashg** [comment]
If we know the types of the usage (e.g. the format for the destination buffers for clear), we can just convert to the correct type precisely. I don't think any of these entrypoints are so hot that this is an impediment.

FWIW, as far as JS is concerned, ints /are/ doubles anyway.

File: spec/index.bs (hunk 1 above)
Comment (kainino0x) [reply spec/index.bs]: IIRC that's not valid; the bindings resolution rules can't choose an overload based based on the type. If an app passes [4,4,4,4], the bindings can't choose which C++ overload to call (the int one or the float one).

**Comment by kainino0x** [comment]
Rebased over #365.

**Pull request merged**